### ISSUE TITLE
Scheduled etcd snapshots are not uploaded to S3, only manual ones

### ISSUE THREAD

**Issue by thewilli** [k3s-io/k3s#7568]
**Environmental Info:**
K3s Version: 

```
k3s version v1.26.4+k3s1 (8d0255af)
go version go1.19.8
```

Node(s) CPU architecture, OS, and Version: 
```
Linux k8s01 5.15.0-71-generic #78-Ubuntu SMP Tue Apr 18 09:00:29 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

Cluster Configuration:
* 3 server nodes

**Describe the bug:**
When I configure an S3 target for etcd snapshots, I can confirm the setting is actually working by creating a manual snapshot via `k3s etcd-snapshot save`. Still, for the scheduled snapshots (`etcd-snapshot-schedule-cron` setting) the snapshots are not uploaded to the S3 bucket and remain local. 

**Steps To Reproduce:**

- Install k3s using HA etcd setup
- Add the following configuration and restart k3s afterwards

```yaml
# /etc/rancher/k3s/config.yaml.d/etcd-backup.yaml
etcd-s3: true
etcd-s3-access-key: <my-access-key>
etcd-s3-secret-key: <my-secret-key>
etcd-s3-bucket: <my-etcd-backup-bucket>
etcd-snapshot-schedule-cron: "0 /6 * * *"
```

**Expected behavior:**
- Manual etcd snapshots (`k3s etcd-snapshot save`) are uploaded to configured s3 bucket
- Scheduled etcd snapshots are uploaded to configured s3 bucket

**Actual behavior:**
- Manual etcd snapshots (`k3s etcd-snapshot save`) are uploaded to configured s3 bucket
- Scheduled etcd snapshots are **not** uploaded to configured s3 bucket and remain local

**Comment by brandond** [comment]
I'm unable to reproduce this. What do the logs say when taking scheduled snapshots?

**Comment by jonwilliams84** [comment]
I have the same issue. My cluster (3 controllers, 5 worker nodes) was already initialised (originally k3s v1.19 and upgraded through each major version to 1.27.3).
I recently set up a minio instance and added the following to `/etc/rancher/k3s/config.yaml`:
```
etcd-snapshot-schedule-cron: " */1 * * *"
etcd-s3: true
etcd-s3-endpoint: minio.REDACTED.com
etcd-s3-access-key: REDACTED
etcd-s3-secret-key: REDACTED
etcd-s3-bucket: kubernetes-etcd-backup
etcd-s3-region: eu-west-2
```

I have also modified the k3s systemd service ExecStart to  this:

`ExecStart=/usr/local/bin/k3s server --data-dir /var/lib/rancher/k3s --disable servicelb --write-kubeconfig-mode 644 --server https://10.32.100.12:6443 --token REDACTED::server:REDACTED --etcd-s3 --etcd-s3-endpoint minio.REDACTED.com --etcd-s3-access-key REDACTED --etcd-s3-secret-key REDACTED --etcd-s3-bucket kubernetes-etcd-backup --etcd-s3-region eu-west-2 --etcd-snapshot-schedule-cron '0 */1 * * *'`

And bounced all 3 of the k3s servers.

Running `sudo k3s etcd-snapshot save` results in:
```
WARN[0000] Unknown flag --etcd-snapshot-schedule-cron found in config.yaml, skipping
INFO[0000] Managed etcd cluster bootstrap already complete and initialized
INFO[0000] Applying CRD helmcharts.helm.cattle.io
INFO[0000] Applying CRD helmchartconfigs.helm.cattle.io
INFO[0000] Applying CRD addons.k3s.cattle.io
INFO[0000] Creating k3s-supervisor event broadcaster
INFO[0000] Saving etcd snapshot to /var/lib/rancher/k3s/server/db/snapshots/on-demand-elt-k8s-3-1690300712
{"level":"info","ts":"2023-07-25T15:58:32.258Z","caller":"snapshot/v3_snapshot.go:65","msg":"created temporary db file","path":"/var/lib/rancher/k3s/server/db/snapshots/on-demand-elt-k8s-3-1690300712.part"}
{"level":"info","ts":"2023-07-25T15:58:32.260Z","logger":"client","caller":"v3@v3.5.7-k3s1/maintenance.go:212","msg":"opened snapshot stream; downloading"}
{"level":"info","ts":"2023-07-25T15:58:32.260Z","caller":"snapshot/v3_snapshot.go:73","msg":"fetching snapshot","endpoint":"https://127.0.0.1:2379"}
{"level":"info","ts":"2023-07-25T15:58:32.447Z","logger":"client","caller":"v3@v3.5.7-k3s1/maintenance.go:220","msg":"completed snapshot read; closing"}
{"level":"info","ts":"2023-07-25T15:58:32.475Z","caller":"snapshot/v3_snapshot.go:88","msg":"fetched snapshot","endpoint":"https://127.0.0.1:2379","size":"41 MB","took":"now"}
{"level":"info","ts":"2023-07-25T15:58:32.476Z","caller":"snapshot/v3_snapshot.go:97","msg":"saved","path":"/var/lib/rancher/k3s/server/db/snapshots/on-demand-elt-k8s-3-1690300712"}
INFO[0001] Saving etcd snapshot on-demand-elt-k8s-3-1690300712 to S3
INFO[0001] Checking if S3 bucket kubernetes-etcd-backup exists
INFO[0001] S3 bucket kubernetes-etcd-backup exists
INFO[0001] Uploading snapshot /var/lib/rancher/k3s/server/db/snapshots/on-demand-elt-k8s-3-1690300712 to S3
INFO[0001] S3 upload complete for on-demand-elt-k8s-3-1690300712
INFO[0001] Reconciling etcd snapshot data in k3s-etcd-snapshots ConfigMap
INFO[0001] Reconciliation of snapshot data in k3s-etcd-snapshots ConfigMap complete
```

Scheduled snapshots however only appear hourly on disk:
```
/var/lib/rancher/k3s/server/db/snapshots
total 405436
drwx------ 2 root root     4096 Jul 25 16:00 .
drwx------ 4 root root     4096 Jul 24 15:46 ..
-rw------- 1 root root 53616672 Jul 25 12:00 etcd-snapshot-elt-k8s-1-1690286402
-rw------- 1 root root 35315744 Jul 25 13:00 etcd-snapshot-elt-k8s-1-1690290003
-rw------- 1 root root 60551200 Jul 25 14:00 etcd-snapshot-elt-k8s-1-1690293603
-rw------- 1 root root 50847776 Jul 25 15:00 etcd-snapshot-elt-k8s-1-1690297204
-rw------- 1 root root 42434592 Jul 25 16:00 etcd-snapshot-elt-k8s-1-1690300801
```

Not sure if this is related or another issue but calling the adhoc snapshot save from each of the servers result in snapshots of different sizes:

![image](https://github.com/k3s-io/k3s/assets/24478629/adeeddc9-b8bb-44a7-a114-619224c63f54)

They all verify fine so not sure if this maybe related to running the snapshots manually on each server - resulting in the etcd-snapshot config map being updated between each of nodes actually saving the snapshot.

**Comment by jonwilliams84** [comment]
Found my issue. For some reason I hadn't updated my endpoint URL correctly. It's now working as expected/designed.

Re the 3 simultaneous snapshots resulting I. Differing sizes, I realise this is not an issue. However, that said; to reduce the load on the minio server while still snapshotting every 2 hours, I have staggered my Cron across the 3 servers:

"0 */6 * * *" 
"0 2/6 * * *"
"0 4/6 * * *"

This means I do not end up with all 3 servers snapshotting simultaneously resulting in 3 files/objects every 2 hours.
